sdk/python/feast/infra/aws_dynamo_provider.py

```diff
@@ -0,0 +1,201 @@
+                )
+            except ClientError as ce:
+                print(ce)
+                if ce.response["Error"]["Code"] == "ResourceNotFoundException":
+                    table = dynamodb.Table(table_name.name)
+
+        for table_name in tables_to_delete:
+            table = dynamodb.Table(table_name.name)
+            table.delete()
+
+    def teardown_infra(
+        self, project: str, tables: Sequence[Union[FeatureTable, FeatureView]]
+    ) -> None:
+        dynamodb = self._initialize_dynamodb()
+
+        for table_name in tables:
+            table = dynamodb.Table(table_name)
+            table.delete()
+
+    def online_write_batch(
+        self,
+        project: str,
+        table: Union[FeatureTable, FeatureView],
+        data: List[
+            Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+        ],
+        progress: Optional[Callable[[int], Any]],
+    ) -> None:
+        dynamodb = self._initialize_dynamodb()
+
+        table_instance = dynamodb.Table(table.name)
+        with table_instance.batch_writer() as batch:
+            for entity_key, features, timestamp, created_ts in data:
+                document_id = compute_datastore_entity_id(entity_key)  # TODO check id
+                # TODO compression encoding
+                batch.put_item(
+                    Item={
+                        "Row": document_id,  # PartitionKey
+                        "Project": project,  # SortKey
+                        "event_ts": str(utils.make_tzaware(timestamp)),
+                        "values": {
+                            k: v.SerializeToString()
+                            for k, v in features.items()  # Serialized Features
+                        },
+                    }
+                )
+
+    def online_read(
+        self,
+        project: str,
+        table: Union[FeatureTable, FeatureView],
+        entity_keys: List[EntityKeyProto],
+    ) -> List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]]:
+        dynamodb = self._initialize_dynamodb()
+
+        result: List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]] = []
+        for entity_key in entity_keys:
+            table_instace = dynamodb.Table(table.name)
+            document_id = compute_datastore_entity_id(entity_key)  # TODO check id
+            response = table_instace.get_item(
+                Key={"Row": document_id, "Project": project}
+            )
+            value = response["Item"]
+
+            if value is not None:
+                res = {}
+                for feature_name, value_bin in value["values"].items():
+                    val = ValueProto()
+                    val.ParseFromString(value_bin.value)
+                    res[feature_name] = val
+                result.append((value["event_ts"], res))
+            else:
+                result.append((None, None))
+        return result
+
+    def materialize_single_feature_view(
```

Comment:
@woop This method was duplicated from GCP, should we make abstract Provider because this part is very well generalized already or this is fine to keep it as is for now and then refactor?

Reply:
Can we do that in a separate PR perhaps? I think we may need to do a code cleanup at some point for generalized functionality.